lib/logstash/inputs/snmptrap.rb

```diff
@@ -72,26 +92,34 @@ def build_trap_listener
       traplistener_opts.merge!({:MibDir => @yamlmibdir, :MibModules => @yaml_mibs})
     end
     @logger.info("It's a Trap!", traplistener_opts.dup)
-    @snmptrap = SNMP::TrapListener.new(traplistener_opts)
+    SNMP::TrapListener.new(traplistener_opts)
   end
 
   def snmptrap_listener(output_queue)
-    build_trap_listener
+    @snmptrap = build_trap_listener
 
     @snmptrap.on_trap_default do |trap|
       begin
-        event = LogStash::Event.new("message" => trap.inspect, "host" => trap.source_ip)
-        decorate(event)
-        trap.each_varbind do |vb|
-          event.set(vb.name.to_s, vb.value.to_s)
-        end
-        @logger.debug("SNMP Trap received: ", :trap_object => trap.inspect)
-        output_queue << event
-      rescue => event
-        @logger.error("Failed to create event", :trap_object => trap.inspect)
+        output_queue << process_trap(trap)
+      rescue => e
+        @logger.error("Failed to create event", :exception => e, :backtrace => e.backtrace, :trap_object => trap)
       end
     end
-    @snmptrap.join
+    @snmptrap
   end # def snmptrap_listener
 
+  def process_trap(trap)
+    @logger.debug? && @logger.debug("SNMP Trap received: ", :trap_object => trap)
```

Comment:
previous code sent `Object#inspect` to ensure we logged a string. Is this change intentional?

```suggestion
    @logger.debug? && @logger.debug("SNMP Trap received: ", :trap_object => trap.inspect)
```